app/code/Magento/Theme/Test/Unit/Controller/Result/JsFooterPluginTest.php

```diff
@@ -0,0 +1,164 @@
+    {
+        $this->scopeConfigMock = $this->getMockBuilder(ScopeConfigInterface::class)
+            ->setMethods(['isSetFlag'])
+            ->disableOriginalConstructor()
+            ->getMockForAbstractClass();
+
+        $this->httpMock = $this->createMock(Http::class);
+
+        $objectManager = new ObjectManagerHelper($this);
+        $this->plugin = $objectManager->getObject(
+            JsFooterPlugin::class,
+            [
+                'scopeConfig' => $this->scopeConfigMock
+            ]
+        );
+    }
+
+    /**
+     * Data Provider for testBeforeSendResponse()
+     *
+     * @return array
+     */
+    public function sendResponseDataProvider(): array
+    {
+        return [
+            [
+                "content" => "<body><h1>Test Title</h1>" .
+                    "<script type=\"text/x-magento-init\">test</script>" .
+                    "<script type=\"text/x-magento-template\">test</script>" .
+                    "<p>Test Content</p></body>",
+                "flag" => true,
+                "result" => "<body><h1>Test Title</h1>" .
+                    "<script type=\"text/x-magento-template\">test</script>" .
+                    "<p>Test Content</p>" .
+                    "<script type=\"text/x-magento-init\">test</script>" .
+                    "\n</body>"
+            ],
+            [
+                "content" => "<body><p>Test Content</p></body>",
+                "flag" => false,
+                "result" => "<body><p>Test Content</p></body>"
+            ],
+            [
+                "content" => "<body><p>Test Content</p></body>",
+                "flag" => true,
+                "result" => "<body><p>Test Content</p>\n</body>"
+            ]
+        ];
+    }
+
+    /**
+     * Test beforeSendResponse
+     *
+     * @param string $content
+     * @param bool $isSetFlag
+     * @param string $result
+     * @return void
+     * @dataProvider sendResponseDataProvider
+     */
+    public function testBeforeSendResponse($content, $isSetFlag, $result): void
+    {
+        $this->httpMock->expects($this->once())
+            ->method('getContent')
+            ->willReturn($content);
+
+        $this->scopeConfigMock->expects($this->once())
+            ->method('isSetFlag')
+            ->with(
+                self::STUB_XML_PATH_DEV_MOVE_JS_TO_BOTTOM,
+                ScopeInterface::SCOPE_STORE
+            )
+            ->willReturn($isSetFlag);
+
+        $this->httpMock->expects($this->any())
+            ->method('setContent')
+            ->with($result);
+
+        $this->plugin->beforeSendResponse($this->httpMock);
+    }
+
+    /**
+     * Data Provider for testBeforeSendResponseIfGetContentIsNotAString()
+     *
+     * @return array
+     */
+    public function ifGetContentIsNotAStringDataProvider(): array
+    {
+        return [
+            [
+                'content' => []
+            ],
+            [
+                'content' => NULL
+            ]
```

Comment:
You can replace this with:
```php
'empty_array' => ['content' => []],
'null' => ['content' => NULL]
```

As a result - when one of them fail - You'll know on which data set it failed.
Also changes to Test Method needed then.

Reply:
@lbajsarowicz Thanks for the feedback, I have updated the keys.

Reply:
@lbajsarowicz Thanks for the feedback, I have updated the keys.